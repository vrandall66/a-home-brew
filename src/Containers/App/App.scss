.App {
  display: grid;
  grid-template-columns: 20% auto;
  grid-template-rows: 22% auto;
  grid-template-areas:
    'nav header header'
    'nav container container'
    'nav container container';
  height: 100vh;
  text-align: center;
}

* {
  background-color: rgba(255, 0, 0, 0.2);
}
* * {
  background-color: rgba(0, 255, 0, 0.2);
}
* * * {
  background-color: rgba(0, 0, 255, 0.2);
}
* * * * {
  background-color: rgba(255, 0, 255, 0.2);
}
* * * * * {
  background-color: rgba(0, 255, 255, 0.2);
}
* * * * * * {
  background-color: rgba(255, 255, 0, 0.2);
}
* * * * * * * {
  background-color: rgba(255, 0, 0, 0.2);
}
* * * * * * * * {
  background-color: rgba(0, 255, 0, 0.2);
}
* * * * * * * * * {
  background-color: rgba(0, 0, 255, 0.2);
}
