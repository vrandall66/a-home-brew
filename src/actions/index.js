export const setBeers = beers => ({
  type: 'SET_BEERS',
  beers
});

export const addProperties = beers => ({
  type: 'ADD_PROPERTIES',
  beers
});

export const setLoading = bool => ({
  type: 'IS_LOADING',
  bool
});

export const setError = error => ({
  type: 'HAS_ERROR',
  error
});

export const toggleBookmark = beer => ({
  type: 'TOGGLE_BOOKMARK',
  beer
});

export const setBookmark = beer => ({
  type: 'SET_BOOKMARK',
  beer
});

export const removeBookmark = beer => ({
  type: 'REMOVE_BOOKMARK',
  beer
});

export const addCurrentlyBrewing = beer => ({
  type: 'ADD_CURRENTLY_BREWING',
  beer
});

export const resetFavorites = () => ({
  type: 'RESET_FAVORITES'
});
