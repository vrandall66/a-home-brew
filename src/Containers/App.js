import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getPage1Beers, getBeerByName } from '../apiCalls/apiCalls';
import { setBeers } from '../actions/index';
import './App.css';
import { bindActionCreators } from 'redux';

export class App extends Component {
  componentDidMount = async () => {
    const { setBeers } = this.props;
    try {
      let beerData = await getPage1Beers();
      setBeers(beerData);
      console.log('1', beerData);
    } catch ({ message }) {
      console.log(message);
    }
  };

  render() {
    return <div className='App'></div>;
  }
}

export const mapStateToProps = ({ beers }) => ({
  beers
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      setBeers,
      getBeerByName
    },
    dispatch
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
