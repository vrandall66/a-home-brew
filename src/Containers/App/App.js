import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { Nav } from '../../Components/Nav/Nav';
import { Header } from '../../Components/Header/Header';
import { Container } from '../Container/Container';
import { BeerPage } from '../BeerPage/BeerPage';
import { getAllBeers } from '../../apiCalls/apiCalls';
import { setBeers } from '../../actions/index';
import { bindActionCreators } from 'redux';
import { PropTypes } from 'prop-types';
import './App.scss';

export class App extends Component {
  componentDidMount = async () => {
    const { setBeers } = this.props;
    try {
      let beerData = await getAllBeers([], 1);
      setBeers(beerData);
    } catch ({ message }) {
      console.log(message);
    }
  };

  renderBeersFromList = list => {
    const { beers } = this.props;
    let renderable = beers.reduce((selectedBeers, beer) => {
      if (list.includes(beer.id)) {
        selectedBeers.push(beer);
      }
      return selectedBeers;
    }, []);
    return renderable;
  };

  render() {
    const { beers, bookmarks, previousBrews, searchResults } = this.props;
    return (
      <div className='App'>
        <Switch>
          <Route
            exact
            path='/'
            render={() => (
              <>
                <Header />
                <Nav />
                <Container
                  beers={beers}
                  type={'beers'}
                  bookmarks={bookmarks}
                  previous={previousBrews}
                />
              </>
            )}
          />
          <Route
            exact
            path='/bookmarked'
            render={() => (
              <>
                <Header />
                <Nav />
                <Container
                  beers={this.renderBeersFromList(bookmarks)}
                  type={'bookmarked'}
                  bookmarks={bookmarks}
                  previous={previousBrews}
                />
              </>
            )}
          />
          <Route
            exact
            path='/previously_brewed'
            render={() => (
              <>
                <Header />
                <Nav />
                <Container
                  beers={this.renderBeersFromList(previousBrews)}
                  type={'previously_brewed'}
                  bookmarks={bookmarks}
                  previous={previousBrews}
                />
              </>
            )}
          />
          <Route
            exact
            path='/(beers|bookmarked|previously_brewed|search)/:id'
            render={({ match }) => {
              const beerDetails = beers.find(
                beer => beer.id === parseInt(match.params.id)
              );
              return (
                <>
                  <Header />
                  <Nav />
                  <BeerPage beerDetails={beerDetails} />
                </>
              );
            }}
          />
          <Route
            exact
            path='/search'
            render={() => (
              <>
                <Header />
                <Nav />
                <Container
                  beers={searchResults}
                  type={'search'}
                  bookmarks={bookmarks}
                  previous={previousBrews}
                />
              </>
            )}
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = ({
  beers,
  bookmarks,
  previousBrews,
  searchResults
}) => ({
  beers,
  bookmarks,
  previousBrews,
  searchResults
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ setBeers }, dispatch);
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);

App.propTypes = {
  beers: PropTypes.array,
  bookmarks: PropTypes.array,
  previousBrews: PropTypes.array,
  searchResults: PropTypes.array,
  setBeers: PropTypes.func
};
