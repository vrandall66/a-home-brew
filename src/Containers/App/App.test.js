import React from 'react';
import { shallow } from 'enzyme';
import { setBeers } from '../../actions'
import { App, mapStateToProps, mapDispatchToProps } from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an object with beers, bookmarks, previousBrews and searchResults arrays', () => {
    const mockStoreState = {
      beers: [{}, {}, {}],
      bookmarks: [1, 2],
      previousBrews: [5],
      searchResults: [{}]
    };

    const expected = {
      beers: [{}, {}, {}],
      bookmarks: [1, 2],
      previousBrews: [5],
      searchResults: [{}]
    };

    const mappedProps = mapStateToProps(mockStoreState);

    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('calls dispatch with setBeers', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setBeers([{}, {}, {}]);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setBeers([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
