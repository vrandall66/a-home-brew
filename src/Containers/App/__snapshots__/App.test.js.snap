// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/bookmarked"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/previously_brewed"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/(beers|bookmarked|previously_brewed|search)/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/search"
      render={[Function]}
    />
  </Switch>
</div>
`;
